### DISCUSSION TITLE
Wazuh Alert Tweaking

### DISCUSSION THREAD

**Discussion by Cra5hedC0w** [Security-Onion-Solutions/securityonion#2598] [Unsupported Versions] [answered]
In the documentation for 2.3.21, for tweaking rules for Wazuh, it says: 

"You can add new rules and modify existing rules in /opt/so/rules/hids/local_rules.xml."

When I cat this file, I see (no actual rules): 
```

<!-- Local rules -->

<!-- Modify it at your will. -->
<!-- Copyright (C) 2015-2020, Wazuh Inc. -->

<!-- Example -->
<group name="local,syslog,sshd,">

  <!--
  Dec 10 01:02:02 host sshd[1234]: Failed none for root from 1.1.1.1 port 1066 ssh2
  -->
  <rule id="100001" level="5">
    <if_sid>5716</if_sid>
    <srcip>1.1.1.1</srcip>
    <description>sshd: authentication failed from IP 1.1.1.1.</description>
    <group>authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,</group>
  </rule>

</group>

```
I take it this is not correct, and is no longer applicable? If this is the case, what is the proper way to adjusting rules that already exist, for example, to whitelist them?

**Comment by trb143** [comment, marked as answer]
See https://documentation.wazuh.com/4.0/learning-wazuh/replace-stock-rule.html for details.  The files is empty to start with as there and no over rides initially.

**Reply by Cra5hedC0w** [reply]
Thank you for the guide, however there is a small issue, where the directory "rules" does not exist where it is said to: 
```
[root@host ruleset]# pwd
/var/ossec/ruleset
[root@host ruleset]# ls
sca
[root@host ruleset]#



```
Where are the original rules located, if not here?  The Wazuh documentation shows it as being in: /var/ossec/ruleset/rules/[rulename].xml
but that directory is not present.

**Reply by weslambert** [reply]
The rules for the server will be located in `/nsm/wazuh`, because we run an agent on the machine itself (`/var/ossec`), and the server's config is mounted through Docker to `/nsm/wazuh`

**Reply by Cra5hedC0w** [reply]
Thank you for the info, I can see the rules in /nsm/wazuh . Would the above article posted by @trb143 still apply with tweaking/modifying rules, even though the wazuh rules are in a different directory, or do I need to change a configuration elsewhere?

**Reply by trb143** [reply]
Yes as the rules are for Wazuh.  Security Onion has just moved the files to a different location.  The behavior of the software would be the same.

**Reply by Cra5hedC0w** [reply]
Thanks for the assist.

**Reply by Cra5hedC0w** [reply]
One more question. For the alerts, I see where we can modify the rules inside each XML (copying over, tweaking, etc). However if I wanted to completely disable a single rule ID inside a ruleset file, would this be just as simple as commenting out the XML for that rule id? For example (random rule inside of 0030-postfix_rules.xml): 

`
`
The documentation points to modifing rules, and even disabling rulesets (.xml), but nothing about disabling a specific rule ID inside a ruleset.

**Reply by trb143** [reply]
Change the inital tag to add overwrite="yes" and then make any other required changes as per the Wazuh documentation.
<rule id="3355" level="0" frequency="$POSTFIX_FREQ" timeframe="120" overwrite="yes">

**Reply by Cra5hedC0w** [reply]
I see where it says overwrite = yes. That makes sense, but nothing discusses disabling the ID, it only talks about tweaking the rule (to maybe detect something different).  Nothing about disabling. I appreciate the assistance, but instead of pointing me to the documentation that doesn't discuss what I am asking,  an example may help.

**Reply by trb143** [reply]
You disable it by changing the logging level so the rule does not fire.

**Reply by Cra5hedC0w** [reply]
Much better, thank you. I see that levels aren't really described in terms of what level 5 would do vs. level 6, or 7. However, I do see that the Wazuh agent has a default configuration of alerting past a certain level by default. Additionally adjusting the threshold may be something that will work for what I am trying to accomplish. 

https://documentation.wazuh.com/3.7/user-manual/manager/alert-threshold.html
https://documentation.wazuh.com/3.11/user-manual/ruleset/ruleset-xml-syntax/rules.html
